util/dockerutil/progress.go

```diff
@@ -1,47 +1,76 @@
 package dockerutil
 
 import (
-	"errors"
+	"encoding/json"
 	"io"
 	"time"
 
 	"github.com/docker/buildx/util/progress"
-	"github.com/docker/cli/cli/streams"
 	"github.com/docker/docker/pkg/jsonmessage"
 	"github.com/moby/buildkit/client"
-	"github.com/moby/buildkit/identity"
-	"github.com/opencontainers/go-digest"
 )
 
-func fromReader(w progress.Writer, name string, rc io.ReadCloser) error {
-	dgst := digest.FromBytes([]byte(identity.NewID()))
-	tm := time.Now()
+const minTimeDelta = 5 * time.Second
```

Comment:
1 or 2 is fine in here. 5sec already feels like something might be stuck if there are not many layers.